src/pyhf/__init__.py

```diff
@@ -27,44 +27,77 @@ def get_backend():
 
 
 @events.register('change_backend')
-def set_backend(backend, custom_optimizer=None):
+def set_backend(backend, custom_optimizer=None, custom_backend=False, _session=None):
     """
     Set the backend and the associated optimizer
 
     Example:
         >>> import pyhf
         >>> import tensorflow as tf
-        >>> pyhf.set_backend(pyhf.tensor.tensorflow_backend(session=tf.Session()))
+        >>> sess = tf.compat.v1.Session()
+        >>> pyhf.set_backend("tensorflow", _session=sess)
 
     Args:
-        backend: One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        backend (`str` or `pyhf.tensor` backend): One of the supported pyhf backends: NumPy, TensorFlow, and PyTorch
+        custom_optimizer (`pyhf.optimize` optimizer): Optional custom optimizer defined by the user
+        custom_backend (`bool`): If :code:`True` the :code:`backend` argument is allowed to be a custom user defined backend outside of the supported backends
+        _session (|tf.compat.v1.Session|_): TensorFlow v1 compatible Session to use when the :code:`"tensorflow"` backend API is used
+
+    .. |tf.compat.v1.Session| replace:: ``tf.compat.v1.Session``
+    .. _tf.compat.v1.Session: https://www.tensorflow.org/api_docs/python/tf/compat/v1/Session
 
     Returns:
         None
     """
     global tensorlib
     global optimizer
 
+    supported_backend_types = (
+        tensor.numpy_backend,
+        tensor.tensorflow_backend,
+        tensor.pytorch_backend,
+    )
+    supported_backend_names = (backend().name for backend in supported_backend_types)
+
+    if isinstance(backend, str) and (
+        backend in supported_backend_names or backend == "numpy_minuit"
+    ):
+        if backend == "numpy":
```

Comment:
CodeFactor is complaining about a complex if/else conditional here. This can be replaced with fewer lines:

```python
backend = getattr(pyhf.tensor, '{0:s}_backend'.format(backend))
if not backend:
  raise ValueError("'{0:s}' is not a supported backend....")
```

Reply:
For some reason I was thinking that it would be important to support "numpy_minuit". I don't know why I thought that as it isn't, not is the name really good. So yes, this is the better way to do things if I ditch it. Thanks! :)